### DISCUSSION TITLE
It would be nice if we could grab as many links as we like and download them.

### DISCUSSION THREAD

**Discussion by audiogamer22** [nianeyna/ao3downloader#159] [Ideas]
I think it would be amazing if I could just find a bunch of links, paste them into a text file and have the app download them one after the other.

There's an app called somethinglike fan fiction downloader that allows you to do this and that's what gave me the idea.

It would honestly make things a bit easier than going through the options pasting the link, doing it again and again.

For now I open 2 apps at once but being able to paste a bunch of links and just sitting back and letting it do its thing would be great.

One more thing, does your app come with an auto updater? If not, it should, it would make things a lot easier on us to not have to go to your website every month or GitHub to check on things.

**Comment by nianeyna** [comment]
the "download links from file" option should be what you want! although now that I think about it it may only work for work and series links and not listing pages like a search result etc... I should check that and make sure it works for those

As for an auto updater, it's a bit difficult to do with a python script, but, hm, maybe I could at least do some sort of *alert* that there's a new version available? Let me think about this.

**Comment by audiogamer22** [comment]
I completely forgot that was an option!

This link might help but I'm not sure as i'm not a programmer.

https://gist.github.com/gesquive/8363131

**Comment by audiogamer22** [comment]
I'm trying it and it seams to be working so far!

I've tried a bookmarks link and a works link? It seams to be working with no issue.

I'm blind and was wondering something, when you paste a link and it starts downloading, do you see on the screen 10 percent of x amount of files downloaded? Because for my screen reader it just reads as blank, unless AO3 wants a break.

Hmm, perhaps you could add a sound once the stuff is completely downloaded? That way we don't have to alt tab and check on the download by pressing the up or down arrow, that's how I do it anyway.


Maybe a little tick sound for waiting and another for done?

It couldn't be to loud though and you'd have to be able to turn it on and off for people that don't want it, what do you think?

**Comment by nianeyna** [comment]
There are one or two functions that don't have any progress indicators (the marked for later download function comes to mind) but the download from file option *should* have an indicator. I can't think of any reason why it wouldn't be legible to your screen reader - I'll look into it.

The sound idea is super cool, I've created issue #161 to track it. I've also created #160 to figure out a solution for updating the script.

**Comment by audiogamer22** [comment]
Well when OCRing the screen, this is not the normal way you navigate using a screen reader btw, this is what i get resuming at: 19:26:35
resuming execution
finished getting page 9. starting page 10
finished getting page 10. starting page
a03 has requested a 300 second break
paused at: 19:27:48
resuming at: 19:32:48
resuming execution
a03 has requested a 11 second break
paused at: 19:33:08
resuming at: 19:33:19
resuming execution
finished getting page 11. starting page
finished getting page 12. starting page
a03 has requested a 300 second break
paused at: 19:34:34
resuming at: 19:39:34
resuming execution
finished getting page 13. starting page
a03 has requested a 14 second break
paused at: 19:39:45
resuming at: 19:39:59
resuming execution
finished getting page 14. starting page
a03 has requested a 300 second break
paused at: 19:41:43
resuming at: 19:46:43
resuming execution
finished getting page 15.
starting page
11
12
13
14
15
16

I'm not sure why that can't be read normaly with an up arrow.

download NVDA, it's free: www.nvaccess.org and try it, the voice is robotic, you can change it if you like by going to NVDA, preferences, speech, change synth, and choosing something like the cor one voices.

TO shut the screen reader up, you press control.


Once you've installed it, do the following, go to your folder, press a until you here ao3.cmd and press enter.

Press a or f, follow the prompts, paste your link and press enter, once you get it started downloading, press your up arrow, you'll hear blank, but I wonder if that's what you see on the screen.

**Comment by audiogamer22** [comment]
I don't know if this helps at all but if you wait on the screen,, the getting page x and AO3 is asking for x minutes message does read, but not if you arrow up and or down.

**Comment by audiogamer22** [comment]
As for the sound idea, yeah, it could help tonnes of people who aren't blind as well.
